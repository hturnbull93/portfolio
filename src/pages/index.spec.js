import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import Home from "./index"

describe("Home", () => {
  it("matches snapshot", () => {
    const mathStub = jest.spyOn(global.Math, "random").mockImplementation(() => 0.5)
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
    mathStub.mockRestore()
  })

  it("renders a Layout", () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find("Layout").length).toEqual(1)
  })

  it("has a h1 containing Harry Turnbull", () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find("h1").text()).toEqual("Hi! I'm Harry Turnbull")
  })
})
