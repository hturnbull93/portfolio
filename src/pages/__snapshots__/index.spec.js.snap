// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div
  className="container"
>
  <nav
    className="navbar"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/projects"
        >
          Projects
        </a>
      </li>
    </ul>
  </nav>
  <h1>
    Harry 
    <span>
      Turnbull.
    </span>
  </h1>
  <h3>
    <span>
      Full Stack Developer. 
    </span>
    <span>
      Based in London UK.
    </span>
  </h3>
  <section>
    <div>
      <h2>
        About.
      </h2>
      <div
        aria-hidden="true"
        className="watermark"
      >
        About.
      </div>
    </div>
    <p>
      I am a Full Stack developer, looking for opportunities to continue to learn, challenge myself and work with great people to build useful apps that consumers enjoy using.
    </p>
    <p>
      I'm passionate about the intersection of technology and creativity. I am self-taught and have been working as a Web Designer for the past three years, which has allowed me to gain skills in Front-End design and implementation, as well as UI/UX design.
    </p>
    <p>
      I have recently completed training at
       
      <a
        href="https://makers.tech/"
      >
        Makers Academy
      </a>
       (London's leading programming bootcamp), to learn best practices, tools and methodologies.
    </p>
  </section>
</div>
`;
